apis/config/v1beta1/imageconfig_types.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:images
+
+// ImageConfig is the Schema for the imageconfigs API
+type ImageConfig struct {
```

Comment:
I haven't finished looking through all the changes but let me ask some questions.

* Is there a 1-to-1 mapping between a CassandraDatacenter and ImageConfig? If so how is that enforced?
* If I change one of the images in my ImageConfig will that result in the underlying StatefulSet being updated?
* What is the default behavior if there is no ImageConfig?
* What happens if I delete the ImageConfig?

Reply:
No, ImageConfig is per cass-operator (it's a operator configuration). If you modify it and restart cass-operator, the behavior is the same as having currently newer version of operator (which would have different versions). This is just a static configuration at this point, no reconcile or updates while cass-operator is running.

If you delete ImageConfig, operator will refuse to start.

Reply:
At the moment the kustomize template generates the configuration, both operatorConfig as well as imageConfig. Missing either one would fail the operator creation.

So as long as the user doesn't intentionally override that creation then defaults are created.